src/Microsoft.Identity.Web.TokenAcquisition/DefaultAuthorizationHeaderBoundProvider.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Abstractions;
+
+namespace Microsoft.Identity.Web
+{
+    internal sealed class DefaultAuthorizationHeaderBoundProvider : IAuthorizationHeaderProvider, IAuthorizationHeaderBoundProvider
+    {
+        private readonly IAuthorizationHeaderProvider _authorizationHeaderProvider;
+
+        private readonly ITokenAcquisition _tokenAcquisition;
+
+        /// <summary>
+        /// Initializes a new instance of the DefaultAuthorizationHeaderBoundProvider class using the specified
+        /// authorization header provider.
+        /// </summary>
+        /// <param name="authorizationHeaderProvider">The provider used to supply authorization headers for HTTP requests.</param>
+        /// <param name="tokenAcquisition">The token acquisition service.</param>
+        public DefaultAuthorizationHeaderBoundProvider(
```

Comment:
Let's modify DefaultAuthorizationHeaderProvider rather to implement the `IAuthorizationHeaderProvider<OperationResult<AuthorizationHeaderInformation, AuthorizationHeaderError>>`